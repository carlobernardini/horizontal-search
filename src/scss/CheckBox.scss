@import 'base';

.orizzonte__checkbox--disabled {
    .orizzonte__checkbox-span--last {
        color: #CCC;
    }
    .orizzonte__checkbox-label {
        cursor: not-allowed;

        &:hover {
            .orizzonte__checkbox-span--first {
                border-color: #d0d1d5;
            }
        }
    }
}

.orizzonte__checkbox-input {
    display: none;

    &:checked {
        + .orizzonte__checkbox-label {
            .orizzonte__checkbox-span--first {
                background: $color-a;
                border-color: $color-a;
                animation: wave .4s ease;

                .orizzonte__checkbox-svg {
                    stroke-dashoffset: 0;
                }

                &:before {
                    transform: scale(2.5);
                    opacity: 0;
                    transition: all .4s ease;
                }
            }
        }
    }
}

.orizzonte__checkbox-label {
    margin: auto;
    user-select: none;
    cursor: pointer;

    &:hover {
        .orizzonte__checkbox-span--first {
            border-color: $color-b;
        }
    }
}

.orizzonte__checkbox-span {
    display: inline-block;
    vertical-align: middle;
    transform: translate3d(0,0,0);

    &--first {
        position: relative;
        width: 12px;
        height: 12px;
        border-radius: 3px;
        transform: scale(1);
        vertical-align: middle;
        border: 1px solid #d0d1d5;
        transition: all .2s ease;

        &:before {
            content: "";
            width: 100%;
            height: 100%;
            background: $color-a;
            display: block;
            transform: scale(0);
            opacity: 1;
            border-radius: 50%;
        }
    }

    &--last {
        padding: 5px 0 5px 7px;
        color: #000;
    }
}

.orizzonte__checkbox-svg {
    position: absolute;
    top: 0;
    left: 0;
    fill: none;
    stroke: #ffffff;
    stroke-width: 2;
    stroke-linecap: round;
    stroke-linejoin: round;
    stroke-dasharray: 16px;
    stroke-dashoffset: 16px;
    transition: all .3s ease;
    transition-delay: .1s;
    transform: translate3d(0, 0, 0);
}
