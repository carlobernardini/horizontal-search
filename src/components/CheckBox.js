import React from 'react';
import PropTypes from 'prop-types';
import '../scss/CheckBox.scss';

const CheckBox = ({ id, label, value }) => (
    <div
        className="orizzonte__checkbox"
    >
        <input
            type="checkbox"
            id={ id }
            className="orizzonte__checkbox-input"
            value={ value }
        />
        <label
            htmlFor={ id }
            className="orizzonte__checkbox-label"
        >
            <span
                className="orizzonte__checkbox-span orizzonte__checkbox-span--first"
            >
                <svg
                    className="orizzonte__checkbox-svg"
                    width="12px"
                    height="10px"
                    viewBox="0 0 12 0"
                >
                    <polyline
                        points="1.5 6 4.5 9 10.5 1"
                    />
                </svg>
            </span>
            <span
                className="orizzonte__checkbox-span orizzonte__checkbox-span--last"
            >
                { label }
            </span>
        </label>
    </div>
);

CheckBox.propTypes = {
    /** Internal ID for this checkbox */
    id: PropTypes.string.isRequired,
    /** Label for this checkbox */
    label: PropTypes.string.isRequired,
    /** Value for this checkbox */
    value: PropTypes.string.isRequired
};

export default CheckBox;
