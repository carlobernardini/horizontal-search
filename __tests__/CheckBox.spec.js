import React from 'react';
import CheckBox from '../src/components/CheckBox';

describe('<CheckBox />', () => {
    it('should render a checkbox', () => {
        const wrapper = shallow(
            <CheckBox
                id="testCheckbox"
                label="Test checkbox"
                value="test"
                onChange={ () => {}}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render a disabled checkbox', () => {
        const wrapper = shallow(
            <CheckBox
                id="testCheckbox"
                label="Test checkbox"
                value="test"
                disabled
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render a selected checkbox', () => {
        const wrapper = shallow(
            <CheckBox
                id="testCheckbox"
                label="Test checkbox"
                value="test"
                selected
            />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
