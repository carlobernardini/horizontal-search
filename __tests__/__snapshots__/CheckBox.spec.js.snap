// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> should render a checkbox 1`] = `
<div
  className="orizzonte__checkbox"
>
  <input
    checked={false}
    className="orizzonte__checkbox-input"
    disabled={false}
    id="testCheckbox"
    onChange={[Function]}
    type="checkbox"
    value="test"
  />
  <label
    className="orizzonte__checkbox-label"
    htmlFor="testCheckbox"
  >
    <span
      className="orizzonte__checkbox-span orizzonte__checkbox-span--first"
    >
      <svg
        className="orizzonte__checkbox-svg"
        height="10px"
        viewBox={
          Array [
            0,
            0,
            10,
            10,
          ]
        }
        width="12px"
      >
        <polyline
          points="1.5 6 3.5 9 8 3"
        />
      </svg>
    </span>
    <span
      className="orizzonte__checkbox-span orizzonte__checkbox-span--last"
    >
      Test checkbox
    </span>
  </label>
</div>
`;

exports[`<CheckBox /> should render a disabled checkbox 1`] = `
<div
  className="orizzonte__checkbox orizzonte__checkbox--disabled"
>
  <input
    checked={false}
    className="orizzonte__checkbox-input"
    disabled={true}
    id="testCheckbox"
    onChange={[Function]}
    type="checkbox"
    value="test"
  />
  <label
    className="orizzonte__checkbox-label"
    htmlFor="testCheckbox"
  >
    <span
      className="orizzonte__checkbox-span orizzonte__checkbox-span--first"
    >
      <svg
        className="orizzonte__checkbox-svg"
        height="10px"
        viewBox={
          Array [
            0,
            0,
            10,
            10,
          ]
        }
        width="12px"
      >
        <polyline
          points="1.5 6 3.5 9 8 3"
        />
      </svg>
    </span>
    <span
      className="orizzonte__checkbox-span orizzonte__checkbox-span--last"
    >
      Test checkbox
    </span>
  </label>
</div>
`;

exports[`<CheckBox /> should render a selected checkbox 1`] = `
<div
  className="orizzonte__checkbox"
>
  <input
    checked={true}
    className="orizzonte__checkbox-input"
    disabled={false}
    id="testCheckbox"
    onChange={[Function]}
    type="checkbox"
    value="test"
  />
  <label
    className="orizzonte__checkbox-label"
    htmlFor="testCheckbox"
  >
    <span
      className="orizzonte__checkbox-span orizzonte__checkbox-span--first"
    >
      <svg
        className="orizzonte__checkbox-svg"
        height="10px"
        viewBox={
          Array [
            0,
            0,
            10,
            10,
          ]
        }
        width="12px"
      >
        <polyline
          points="1.5 6 3.5 9 8 3"
        />
      </svg>
    </span>
    <span
      className="orizzonte__checkbox-span orizzonte__checkbox-span--last"
    >
      Test checkbox
    </span>
  </label>
</div>
`;
